busybox/busybox.py

```diff
@@ -29,8 +29,9 @@ def harness(dut: Dut, ctx: TestContext, result: TestResult):
             msg.append("\t\t" + parsed["line"])
             continue
 
-        if subresult:
-            # We ended processing test result and message
+        if idx == 0:
```

Comment:
same as with the mbedtls - msg would be appended to the next subresult instead of the correct one. Similar approach to the one proposed for mbedtls needs to be applied here

(and some testing with tests-failing-on-purpose to ensure messagess are appended to the correct testcase :))